src/mpi/romio/mpi-io/glue/default/mpio_file.c

```diff
@@ -54,7 +54,7 @@ MPI_File MPIO_File_f2c(MPI_Fint fh)
         /* there is no way to return an error from MPI_File_f2c */
         return MPI_FILE_NULL;
     }
-    return ADIOI_Ftable[fh];
+    return (MPI_File) ADIOI_Ftable[fh];
```

Comment:
I would like to get rid of the cast. Can we use a wrapper similar to `MPIO_File_resolve`? Personally I find `_resolve` ambiguous. Maybe `MPIO_File_{to,from}_adio`?

Reply:
As you can see from the original codes, the type casting is done implicitly.
That change of mine can be reversed.

Reply:
Implicit or not, it is still a cast. If we want to use `MPIO_File_resolve` to abstract the conversion from `MPI_File` to `ADIO_File`, then it doesn't make sense not to abstract the conversion from `ADIO_File` to `MPI_File`. If we want to require `MPI_File` to be equivalent to `ADIO_File`, then we should `#define ADIO_File MPI_File` and get rid of all the cast. The cast is just a bad code smell and inconsistent design.

Reply:
It is the best your suggested changes to be a separate PR...

Reply:
Ok, I'll make the PR.